### PR TITLE
Update to Dagger 2.22.1Set selected item

### PR DESCRIPTION
**Pull request by aclassen**
Updated sample app

### CODE REVIEW

File: app/src/main/java/io/stanwood/mhwdb/datasource/net/mhw/MhwArmor.kt [hunk 1]
```diff
@@ -30,6 +29,6 @@ data class MhwArmor(
     val type: String,
     val rank: String,
     val name: String,
-    @Optional val assets: MhwArmorAsset? = null,
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/mhwdb/datasource/net/mhw/MhwArmor.kt]: Aren't optionals supported anymore, or don't we need them anymore?

File: app/src/main/java/io/stanwood/mhwdb/feature/armors/ui/ArmorDetailsFragment.kt [hunk 2]
```diff
@@ -62,7 +64,13 @@ class ArmorDetailsFragment : Fragment(), HasSupportFragmentInjector {
     }
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?) =
-        FragmentArmorDetailsBinding.inflate(inflater, container, false, dataBindingComponent)
+        DataBindingUtil.inflate<FragmentArmorDetailsBinding>(
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/mhwdb/feature/armors/ui/ArmorDetailsFragment.kt]: Is there an advantage in using `DataBindingUtil.inflate()` compared to `*Binding.inflate()`?

File: app/src/main/java/io/stanwood/mhwdb/feature/armors/ui/ArmorsAdapter.kt [hunk 3]
```diff
@@ -38,8 +40,11 @@ class ArmorsAdapter(
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
         when (viewType) {
-            ArmorItem.ArmorViewModel.VIEW_TYPE -> DividerViewHolder(LayoutArmorItemBinding.inflate(
+
+            ArmorItem.ArmorViewModel.VIEW_TYPE -> DividerViewHolder(
+                DataBindingUtil.inflate<LayoutArmorItemBinding>(
                 inflater,
+                    R.layout.layout_armor_item,
```
Comment (ubuntudroid) [inline app/src/main/java/io/stanwood/mhwdb/feature/armors/ui/ArmorsAdapter.kt]: Wrong indentation?

File: nav/src/main/java/io/stanwood/framework/arch/nav/NavigationViewSyncer.kt [hunk 4]
```diff
@@ -38,7 +38,9 @@ class NavigationViewSyncer(
 
     val destinationChangedListener: (NavController, NavDestination, Bundle?) -> Unit = { _, destination, _ ->
         bottomNav?.also {
-            it.selectedItemId = destination.id
+            it.menu.findItem(destination.id)?.apply {
```
Comment (ubuntudroid) [inline nav/src/main/java/io/stanwood/framework/arch/nav/NavigationViewSyncer.kt]: @aclassen Isn't this a straight revert from your last PR? https://github.com/stanwood/framework-arch-android/pull/26

Didn't it work out in the end?

File: app/src/main/java/io/stanwood/mhwdb/datasource/net/mhw/MhwArmor.kt (hunk 1 above)
Comment (aclassen) [reply app/src/main/java/io/stanwood/mhwdb/datasource/net/mhw/MhwArmor.kt]: optional is deprecated

File: nav/src/main/java/io/stanwood/framework/arch/nav/NavigationViewSyncer.kt (hunk 4 above)
Comment (aclassen) [reply nav/src/main/java/io/stanwood/framework/arch/nav/NavigationViewSyncer.kt]: stack overflow

File: app/src/main/java/io/stanwood/mhwdb/feature/armors/ui/ArmorDetailsFragment.kt (hunk 2 above)
Comment (aclassen) [reply app/src/main/java/io/stanwood/mhwdb/feature/armors/ui/ArmorDetailsFragment.kt]: This usage is deprecated

File: app/src/main/java/io/stanwood/mhwdb/datasource/net/mhw/MhwArmor.kt (hunk 1 above)
Comment (ubuntudroid) [reply app/src/main/java/io/stanwood/mhwdb/datasource/net/mhw/MhwArmor.kt]: Sweet! So now we can just have nullable types and thus tell the serializer they are optional? Very good! 😍

File: nav/src/main/java/io/stanwood/framework/arch/nav/NavigationViewSyncer.kt (hunk 4 above)
Comment (ubuntudroid) [reply nav/src/main/java/io/stanwood/framework/arch/nav/NavigationViewSyncer.kt]: @aclassen Do you have a link to the thread? Would be interested in the discussion.

Comment (aclassen) [reply nav/src/main/java/io/stanwood/framework/arch/nav/NavigationViewSyncer.kt]: it results in a stack overflow :D @ubuntudroid

**Pull request merged**